### PR TITLE
enhance(cmd/verify): add goroutine count to improve verify speed

### PR DESCRIPTION
**Pull request by chenminjian**
refs: #5696 

License: MIT
Signed-off-by: chenminjian <727180553@qq.com>

### CODE REVIEW

**Comment by chenminjian** [comment]
hey, @Stebalien . I use `Ctrl + C` to interrupt `ipfs repo verify`, then  IPFS `panic: send on closed channel`. It looks like we can't use `goroutine` and `select` simply. I am going to use `sync.WaitGroup` to avoid it. Does that make sense? Can you give me some advice? Thx a lot.

File: core/commands/repo.go [hunk 1]
```diff
@@ -302,13 +302,35 @@ var repoVerifyCmd = &oldcmds.Command{
...
 		var i int
+		var count int
+		verify := make(chan string)
+		defer close(verify)
+
 		for k := range keys {
-			_, err := bs.Get(k)
-			if err != nil {
+			go func() {
+				_, err := bs.Get(k)
```
Comment (overbool) [inline core/commands/repo.go]: `k` should be passed in by the `go func(k cid.Cid){...}(k)`

File: core/commands/repo.go [hunk 2]
```diff
@@ -302,13 +302,35 @@ var repoVerifyCmd = &oldcmds.Command{
...
 		var fails int
 		var i int
+		var count int
+		verify := make(chan string)
+		defer close(verify)
+
 		for k := range keys {
-			_, err := bs.Get(k)
-			if err != nil {
+			go func() {
```
Comment (Stebalien) [inline core/commands/repo.go]: So, spawning a goroutine per key is going to blow through memory like nothing else. We'll have to:

1. Span N goroutines where N is something like `runtime.NumCPU * 2` (ish?).
2. Have them all read off the keys channel.
3. Have them all write to the same "results" channel.
4. Have them call `Done` on a wait-group on exit.
5. Spin up another goroutine that closes the "results" channel when the wait-group finishes.
6. Outside, have a for loop reading off the results channel.

File: core/commands/repo.go (hunk 1 above)
Comment (chenminjian) [reply core/commands/repo.go]: Thx a lot.

File: core/commands/repo.go (hunk 2 above)
Comment (chenminjian) [reply core/commands/repo.go]: This is great advice. Thx a lot. :)

**Pull request merged**